publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/dynamodbstream/DynamodbStreamToEventBridgeHandler.java

```diff
@@ -32,72 +45,99 @@
     public static final String DYNAMO_DB_STREAM_SOURCE = "DynamoDbStream";
-    private static final String OUTPUT_EVENT_TOPIC = "OUTPUT_EVENT_TOPIC";
-    private final S3Client s3Client;
+    public static final String RECOVERY_QUEUE = new Environment().readEnv("RECOVERY_QUEUE");
+    private static final Logger logger = LoggerFactory.getLogger(DynamodbStreamToEventBridgeHandler.class);
+    private final S3Driver s3Driver;
+    private final QueueClient sqsClient;
     private final EventBridgeClient eventBridgeClient;
 
-    private final String dynamoDbUpdateEventTopic;
-
     @JacocoGenerated
     public DynamodbStreamToEventBridgeHandler() {
-        this(defaultS3Client(), defaultEventBridgeClient(), new Environment().readEnv(OUTPUT_EVENT_TOPIC));
+        this(S3Driver.defaultS3Client().build(), defaultEventBridgeClient(),
+             ResourceQueueClient.defaultResourceQueueClient(RECOVERY_QUEUE));
     }
 
-    protected DynamodbStreamToEventBridgeHandler(S3Client s3Client,
-                                                 EventBridgeClient eventBridgeClient,
-                                                 String dynamoDbUpdateEventTopic) {
-        this.s3Client = s3Client;
+    protected DynamodbStreamToEventBridgeHandler(S3Client s3Client, EventBridgeClient eventBridgeClient,
+                                                 QueueClient sqsClient) {
+        this.s3Driver = new S3Driver(s3Client, EVENTS_BUCKET);
         this.eventBridgeClient = eventBridgeClient;
-        this.dynamoDbUpdateEventTopic = dynamoDbUpdateEventTopic;
+        this.sqsClient = sqsClient;
     }
 
     @Override
-    public Set<PutEventsResponse> handleRequest(DynamodbEvent inputEvent, Context context) {
-        return inputEvent.getRecords()
-                   .stream()
-                   .map(attempt(JsonUtils.dtoObjectMapper::writeValueAsString))
-                   .map(attempt -> attempt.map(this::storeFileInS3Bucket))
-                   .map(attempt -> attempt.map(this::createEvent))
-                   .map(attempt -> attempt.map(eventReference -> sendEvent(eventReference, context)))
-                   .map(Try::orElseThrow)
-                   .collect(Collectors.toSet());
+    public EventReference handleRequest(DynamodbEvent inputEvent, Context context) {
+        var dynamodbStreamRecord = inputEvent.getRecords().getFirst();
+        var dataEntryUpdateEvent = convertToDataEntryUpdateEvent(dynamodbStreamRecord);
+        return dataEntryUpdateEvent.notEmpty() ? sendEvent(dataEntryUpdateEvent, context) : DO_NOT_EMIT_EVENT;
     }
 
-    @JacocoGenerated
-    private static S3Client defaultS3Client() {
-        return S3Driver.defaultS3Client().build();
+    private static SortableIdentifier getIdentifier(DataEntryUpdateEvent blobObject) {
+        return Optional.ofNullable(blobObject.getOldData())
+                   .map(Entity::getIdentifier)
+                   .orElseGet(() -> blobObject.getNewData().getIdentifier());
```

Comment:
Line 83: why does DataEntryUpdateEvent::getOldData and DataEntryUpdateEvent::getNewData not return Optional<Entity>?

Line 84: what happens when this is null, empty, blank?

Line 85: This train wreck might give an NPE too.